From 0d25342c5db4976d0364acd351b2da7f235e7bf6 Mon Sep 17 00:00:00 2001
From: Pradeep Kumar Chitrapu <quic_pradeepc@quicinc.com>
Date: Thu, 12 Dec 2024 13:25:09 -0800
Subject: [PATCH 1/2] QSDK: CP: Enable compilation support for 11S MLO Test
 Framework

Feature can be enabled by selecting option ATH12K_TEST_FW_FOR_11S_MLO
in make menuconfig before starting compilation. This will enable flag
CPTCFG_ATH12K_TEST_FW_FOR_11S_MLO  within ath12k to use.
By default, the compilation flag is disabled.

Patch-dependency: Ifae0e243a05 ("QSDK:CP: Add compile flag for ATH12K Host Test Framework for 11s MLO")

Signed-off-by: Pradeep Kumar Chitrapu <quic_pradeepc@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/Kconfig  | 9 ++++++++-
 drivers/net/wireless/ath/ath12k/Makefile | 1 +
 local-symbols                            | 1 +
 3 files changed, 10 insertions(+), 1 deletion(-)

--- a/drivers/net/wireless/ath/ath12k/Kconfig
+++ b/drivers/net/wireless/ath/ath12k/Kconfig
@@ -92,6 +92,13 @@ config ATH12K_PPE_DS_SUPPORT
 
 		Say Y to enable PPE DS Support. If unsure, say N.
 
+config ATH12K_TEST_FW_FOR_11S_MLO
+	bool "QTI ath12k Test Framework support for 11s MLO"
+	depends on ATH12K_DEBUGFS
+	help
+	 Enable ath12k Test Framework support for 11s MLO.
+	 Say Y to enable Test Framework Support for 11s MLO. If unsure, say N.
+
 config ATH12K_SAWF
 	bool "ath12k Service Aware WiFi Framework and Telemetry support"
 	depends on ATH12K
--- a/drivers/net/wireless/ath/ath12k/Makefile
+++ b/drivers/net/wireless/ath/ath12k/Makefile
@@ -39,6 +39,7 @@ ath12k-$(CPTCFG_ATH12K_PPE_DS_SUPPORT) +
 ath12k-$(CPTCFG_ATH12K_SAWF) += sawf.o telemetry.o telemetry_agent_if.o
 ath12k-$(CPTCFG_ATH12K_CFR) += cfr.o
 ath12k-$(CPTCFG_ATH12K_POWER_BOOST) += power_boost.o
+ath12k-$(CPTCFG_ATH12K_TEST_FW_FOR_11S_MLO) += test_mlo_mesh.o
 
 # for tracing framework to find trace.h
 CFLAGS_trace.o := -I$(src)
--- a/local-symbols
+++ b/local-symbols
@@ -187,6 +187,7 @@ ATH12K_DEBUG=
 ATH12K_DEBUGFS=
 ATH12K_TRACING=
 ATH12K_PPE_DS_SUPPORT=
+ATH12K_TEST_FW_FOR_11S_MLO=
 ATH12K_SPECTRAL=
 ATH12K_PKTLOG=
 ATH12K_CFR=
